### DISCUSSION TITLE
Would more documentation for C++ be appreciated?

### DISCUSSION THREAD

**Discussion by jparsert** [Z3Prover/z3#6566] [Ideas]
Most of the documentation is for the C/Python API. Usually, when using the C++ API I look into the implementation and then look at the documentation of the underlying C functions. I was thinking of gradually adding documentation for the C++ api especially for the most used functions and the functions that differ from the C/Python API (esp. constructors etc.). Would this be appreciated or is there a specific reason a lot of documentation is missing for the C++ api.

**Comment by NikolajBjorner** [comment]
Sure, you are very welcome.
I think the main issue is to save overhead and deal with possible divergence. 
We have been using Doxygen to generate documents. It is built on nightly and you can also build it if installing dependencies for doxygen. 
Since all functionality is based on the C API the semantics of operations can be mentioned there. Links and references 
may be useful.
